handlers.go

```diff
@@ -366,7 +390,10 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 			// add the received addresses to our peerstore.
 			dht.peerstore.AddAddrs(pi.ID, pi.Addrs, peerstore.ProviderAddrTTL)
 		}
-		dht.ProviderManager.AddProvider(ctx, key, p)
+		err := dht.ProviderManager.AddProviderNonBlocking(ctx, key, p)
```

Comment:
Why not do a `mybucket` style check here as well?

Reply:
As far as I understand, we're _always_ in the bucket here, right?